src/ripple/core/DatabaseCon.h

```diff
@@ -177,7 +183,14 @@ class DatabaseCon
     LockedSociSession
     checkoutDb()
```

Comment:
I think this function should take a `beast::Journal j` parameter rather than adding data member `beast::Journal const j_` . Yes that means adding of this parameter everywhere, but it's better than giving `Config` an accessor to `Journal` and then storing that inside `DatabaseCon`. Also, that means the journal will be tied to where the `checkoutDb` is called rather than where `DatabaseCon` is initialised, which is good news for `doVacuumDB` (and perhaps other places as well)

Reply:
NVM, I am starting to think this is not viable. Will have to revisit this at some point later, with proper performance logging/monitoring (no journal).